src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py

```diff
@@ -234,13 +239,18 @@ def build_resource_bicep(self) -> BicepDefinitionElementBuilder:
             # List of NF RET names, for adding to required part of CGS
             nf_names.append(processor.name)
 
+
         template_path = get_template_path(
             NSD_TEMPLATE_FOLDER_NAME, NSD_DEFINITION_TEMPLATE_FILENAME
         )
 
+        # If all NF RETs nfvi_types are AzureCore, only make one nfviFromSite object

```

Comment:
I'd argue this check should come immediately after the for loop, as it's related to the loop, whereas setting the `template_path` is not.

A minor point, and maybe there's a reason you wanted it here, but in general good to keep related operations together.

Reply:
Also, can you explain _why_ we do this. Ideally, the code should be clear on the what, and the comment should explain the why.